subsys/emul/i2c/emul_sbs_gauge.c

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Emulator for SBS 1.1 compliant smart battery fuel gauge.
+ * This emulator is *NOT* fully featured yet, only enough to get the sbs_gauge driver tests passing.
+ */
+
+#define DT_DRV_COMPAT sbs_sbs_gauge
+
+#define LOG_LEVEL CONFIG_SENSOR_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(sbs_sbs_gauge);
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/emul.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/i2c_emul.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/drivers/spi_emul.h>
+#include <zephyr/sys/byteorder.h>
+
+#include <sbs_gauge.h>
+
+/** Run-time data used by the emulator */
+struct sbs_gauge_emul_data {
+	/** sbs_gauge device being emulated */
+	const struct device *dev;
+	/** Current register to read (address) */
+	uint32_t cur_reg;
+};
+
+/** Static configuration for the emulator */
+struct sbs_gauge_emul_cfg {
+	/** I2C address of emulator */
+	uint16_t addr;
+};
+
+static void reg_write(const struct emul *target, int regn, int val)
+{
+	/* Unimplemented reg_write handler */
+	struct sbs_gauge_emul_data *data = target->data;
+
+	ARG_UNUSED(data);
```

Comment:
Nit: `data` isn't an argument. `target` is. Similar concern below.

Reply:
Whoops. Missed that when converting from the old emul api.